ynput/OpenPype
# TVPaint: Use Publisher tool

## Brief description
Use Publisher tool and new creation system in TVPaint integration.

## Description
Using new creation system makes TVPaint integration a little bit easier to maintain for artists. Removed unneeded tools Creator and Subset Manager tools. Goal is to keep the integration work as close as possible to previous integration. Some changes were made but primarilly because they were not right using previous system.

All creators create instance with final family instead of changing the family during extraction. Render passes are not related to group id but to render layer instance. Render layer is still related to group. Workfile, review and scene render instances are created using autocreators instead of auto-collection during publishing. Subset names are fully filled during publishing but instance labels are filled on refresh with the last known right value. Implemented basic of legacy convertor which should convert render layers and render passes.

## Additional info
Because of special subset name template for `render` families I had to add option to filter subset name templates by different family than is family to fill the subset name, which is kind anti-pattern -> fill family is `render` but for filtering is used `renderLayer`, `renderPass` or `renderScene` family. It can be kept as is now but it probably should be changed.
One option is to have same subset name template for all `render` families in TVPaint, which could work but could block other implementations of possible TVPaint render families -> this would break backwards compatibility for subset name templates.

## TODOs
Update documentation.

## Testing notes:
1. Start TVPaint
2. Make a scene
3. Open Publisher
4. Create a Render Layer/s (you can create it out of selection)
5. Reset publisher (this is to update Render Layers combobox)
6. Create a Render Pass/es - Based on selected TVPaint layers in scene
7. Enable/Disable instances
8. Publish

Legacy convertor
1. Open TVPaint file with legacy created instances
2. Run Convert
3. All instances should be converted to new creators
